.github/RELEASE_PROCEDURE.md

```diff
@@ -0,0 +1,39 @@
+# Release Procedure
+
+
+## Tagging The Release
+_This file uses `v3.0.0` as an example release_
+
+1. Checkout a release branch using `git checkout upstream/master -b release-3.0`
+2. Update the `VERSION` [file](../VERSION) to contain `v3.0.0`
+3. Update the `CHANGELOG.md` [file](../CHANGELOG.md) section `Changes Since v2.6.0` -> `v3.0.0 - [YYYY.MM.DD]` and add new section `Changes Since v3.0.0` to track new changes
+4. Make any necessary changes to the `CHANGELOG.md` [file](../CHANGELOG.md) at this time
+5. Commit these changes using `git commit -am "Release v3.0.0"`
+6. Tag the release using `git tag -a -m "Singularity v3.0.0" v3.0.0`
+7. Push the `release-3.0` branch to `upstream` using `git push upstream release-3.0`, also push the `v3.0.0` tag using `git push upstream v3.0.0`
+7. Merge the `upstream/release-3.0` branch into the `upstream/master` branch via a GitHub PR
+8. Checkout a new branch _e.g. development_ using `git checkout upstream/master -b development` 
+9. Update the `VERSION` [file](../VERSION) to contain `v3.1.0-devel`
```

Comment:
`v3.1.0-devel` semantically means a pre-release version of 3.1.0. We should use `v3.0.0+devel` instead in this step, so that we can still programatically reason about the version (for example it being < `v3.1.0-alpha.1`.)